pulp_2to3_migration/app/plugin/content.py

```diff
@@ -416,46 +425,46 @@ def get_remote_by_importer_id(importer_id):
                             pb.increment()
                         continue
 
+                    relative_path = (
+                        pulp_2to3_detail_content.relative_path_for_content_artifact
+                    )
+
                     if is_lazy_type and pulp2lazycatalog:
                         # handle DA and RA creation for content that supports on_demand
                         # Downloaded or on_demand content with LCEs.
                         #
                         # To create multiple remote artifacts, create multiple instances of
                         # declarative content which will differ by url/remote in their
                         # declarative artifacts
-                        at_least_one_lce_migrated = False
+                        remote_lce_tuples = []
+                        deferred_download = not pulp2content.downloaded
                         for lce in pulp2lazycatalog:
                             remote = get_remote_by_importer_id(lce.pulp2_importer_id)
-                            deferred_download = not pulp2content.downloaded
-                            if not remote and deferred_download:
-                                continue
+                            if remote:
+                                remote_lce_tuples.append((remote, lce))
 
-                            relative_path = (
-                                pulp_2to3_detail_content.relative_path_for_content_artifact
-                            )
+                        for remote, lce in remote_lce_tuples:
                             da = DeclarativeArtifact(
                                 artifact=artifact,
                                 url=lce.pulp2_url,
                                 relative_path=relative_path,
                                 remote=remote,
                                 deferred_download=deferred_download)
                             lce.is_migrated = True
-                            at_least_one_lce_migrated = True
                             dc = DeclarativeContent(content=pulp3content, d_artifacts=[da])
+                            future_relations.update({'lces': list(pulp2lazycatalog)})
                             dc.extra_data = future_relations
                             await self.put(dc)
 
-                        if not at_least_one_lce_migrated:
+                        if not remote_lce_tuples and deferred_download:
```

Comment:
Note for myself. Need to cover the case when content has LCE but no related importers are migrated, and content is downloaded, so we can actually proceed. In this implementation it is skipped.